packages/builders/src/jest/jest.builder.ts

```diff
@@ -16,37 +17,66 @@ try {
 const { runCLI } = require('jest');
 
 export interface JestBuilderOptions {
+  codeCoverage?: boolean;
   jestConfig: string;
+  testDirectory?: string;
+  testFile?: string;
+  setupFile?: string;
   tsConfig: string;
-  watch: boolean;
-  bail?: boolean;
+  bail?: boolean | number;
   ci?: boolean;
-  codeCoverage?: boolean;
-  onlyChanged?: boolean;
+  color?: boolean;
+  json?: boolean;
   maxWorkers?: number;
+  onlyChanged?: boolean;
+  outputFile?: string;
   passWithNoTests?: boolean;
   runInBand?: boolean;
-  setupFile?: string;
   silent?: boolean;
-  updateSnapshot?: boolean;
   testNamePattern?: string;
+  updateSnapshot?: boolean;
+  useStderr?: boolean;
+  watch?: boolean;
 }
 
 export default class JestBuilder implements Builder<JestBuilderOptions> {
   run(
     builderConfig: BuilderConfiguration<JestBuilderOptions>
   ): Observable<BuildEvent> {
     const options = builderConfig.options;
+
+    if (options.testFile) {
+      const filePath = options.testDirectory
+        ? path.resolve(options.testDirectory, options.testFile)
+        : realpathSync(options.testFile);
+
+      if (filePath) {
+        const project = this.getProjectForFile(filePath);
+
+        if (project && project.root !== builderConfig.root) {
+          return of({
```

Comment:
Why is this necessary? and If the specified file is not in the project, should it be a failure?